Sources/Plasma/NucleusLib/pnAsyncCoreExe/pnAceSocket.cpp

```diff
@@ -430,6 +442,46 @@ static void HardCloseSocket(AsyncSocket conn)
         LogMsg(kLogError, "Failed to close socket: {}", err.message());
 }
 
+//===========================================================================
+static unsigned SocketCloseTimerCallback (void *) {
+    std::vector<AsyncSocket> sockets;
+
+    unsigned sleepMs;
+    unsigned currTimeMs = TimeGetMs();
+    {
+        hsLockGuard(s_connectCrit);
+
+        for (;;) {
+            // If there are no more sockets pending destruction then
+            // wait forever; the timer will be restarted when the
+            // next socket is queued onto the list.
+            AsyncSocket sock = s_socketList.front();
+            if (!sock) {
+                sleepMs = kAsyncTimeInfinite;
+                break;
+            }
+
+            // Wait until the socket close timer expires
+            if (0 < (signed) (sleepMs = sock->closeTimeMs - currTimeMs))
+                break;
+
+            if (sock->fSock.is_open())
+                sockets.push_back(sock);
+
+            // To avoid a race condition, this unlink must occur
+            // after the socket handle has been cleared
+            s_socketList.pop_front();
+        }

```

Comment:
Should this have a sleep somewhere so that we aren't getting into a loop that consumes an entire CPU core?

Reply:
This code is part of the timer path, which was part of the soft disconnect path, which was removed. This function implementation isn't part of the PR anymore. I did find a declaration of this function that should have been removed, but this implementation shouldn't be/isn't part of the current head.